### PR TITLE
Feat/add service filter nomad sd

### PR DESCRIPTION
**Pull request by quentin9696**
This implement #11465

### CODE REVIEW

**Comment by quentin9696** [comment]
@roidelapluie @LeviHarrison this PR is ready for review

File: discovery/nomad/nomad_test.go [hunk 1]
```diff
@@ -44,6 +44,33 @@ func NewSDMock(t *testing.T) *SDMock {
...
+	}
+}
+
+func TestNonConfiguredServices(t *testing.T) {
+	conf := &SDConfig{}
+	nomadDiscovery, err := NewDiscovery(conf, nil)
+	if err != nil {
+		t.Errorf("Unexpected error when initializing discovery %v", err)
+	}
+	if !nomadDiscovery.shouldWatchService("nonConfiguredServiceName") {
```
Comment (roidelapluie) [inline discovery/nomad/nomad_test.go:69]: please use require.True

File: discovery/nomad/nomad_test.go [hunk 2]
```diff
@@ -44,6 +44,33 @@ func NewSDMock(t *testing.T) *SDMock {
...
+		Services: []string{"configuredServiceName"},
+	}
+	nomadDiscovery, err := NewDiscovery(conf, nil)
+	if err != nil {
+		t.Errorf("Unexpected error when initializing discovery %v", err)
+	}
+	if !nomadDiscovery.shouldWatchService("configuredServiceName") {
+		t.Errorf("Expected service %s to be watched", "configuredServiceName")
+	}
+	if nomadDiscovery.shouldWatchService("nonConfiguredServiceName") {
```
Comment (roidelapluie) [inline discovery/nomad/nomad_test.go:58]: please use require.False

**Comment by roidelapluie** [comment]
I am puzzled by this. Can't we filter the services at the API level?

I'd say if that is not possible at the API level, this should not belong to Prometheus, but be done via relabeling.

**Comment by quentin9696** [comment]
Hi @roidelapluie and thanks you for your review. 

I just use the same logic that what's done on consul service discovery. 

I think it's important to keep some helpers on most common usecase like filter a SD by its name using a single line config instead of doing this multiline config block. 
```yaml
relabel_configs:
  - source_labels: [__meta_nomad_service]
    regex: "openweather"
    action: keep
``` 

Nomad don't provide a way to filter a list of service the API levelm as well as it's done in consul. 

What do you think?

**Comment by roidelapluie** [comment]
Can we use Filter in QueryOptions ?

**Comment by quentin9696** [comment]
@roidelapluie I did some research, and try to use the QueryOptions Filter without sucess. 

I ask my question to hashicorp and here is the reply: https://discuss.hashicorp.com/t/use-query-filter-on-service-list/46656

**Comment by roidelapluie** [comment]
my comments still stand

**Comment by beorn7** [comment]
Looking at this during the bug scrub again.

Since there was no follow-up since the last pass, we'll close this PR for now.

@quentin9696 if you come back working on this, feel free to re-open. We assume the feature request in #11465 is still valid and would welcome any further help with it.

**Pull request closed without merging**